modules/utils/subxt/src/lib.rs

```diff
@@ -215,3 +216,11 @@ pub fn fisherman_storage_key(address: Vec<u8>) -> Vec<u8> {
 
 	[pallet_prefix, storage_prefix, key_1, address.encode()].concat()
 }
+
+pub async fn get_latest_block_hash<C: subxt::Config>(
+	rpc: &LegacyRpcMethods<C>,
+) -> Result<HashFor<C>, anyhow::Error> {
+	rpc.chain_get_block_hash(None)
+		.await?
+		.ok_or_else(|| anyhow!("Failed to query latest block hash"))
+}
```

Comment:
I don't know if we need this tbh

Reply:
@dharjeezy, just use that line directly where it's needed instead of creating this function.
Also in this function https://github.com/polytope-labs/hyperbridge/blob/0f37db9c89cd331a3608566d7f7ef02352338320/tesseract/messaging/messaging/src/lib.rs#L443

Filter out receipts where the request_fee_metadata  or response_fee_metadata on the source chain is zero.